Doc/library/shutil.rst

```diff
@@ -433,23 +433,44 @@ Directory and files operations
    When no *path* is specified, the results of :func:`os.environ` are used,
    returning either the "PATH" value or a fallback of :attr:`os.defpath`.
 
-   On Windows, the current directory is always prepended to the *path* whether
-   or not you use the default or provide your own, which is the behavior the
-   command shell uses when finding executables.  Additionally, when finding the
-   *cmd* in the *path*, the ``PATHEXT`` environment variable is checked.  For
-   example, if you call ``shutil.which("python")``, :func:`which` will search
-   ``PATHEXT`` to know that it should look for ``python.exe`` within the *path*
-   directories.  For example, on Windows::
+   On Windows, the current directory is prepended to the *path* if
+   the *mode* does not include ``os.X_OK``. When the *mode* does include
+   ``os.X_OK``, the Windows API ``NeedCurrentDirectoryForExePathW`` will be
+   consulted to determine if the current directory should be prepended to
+   *path*. To avoid consulting the current working directory for executables,
+   and thereby, match the behavior of non-Windows shells: set the environment
```

Comment:
I wouldn't mention "match the behavior of non-Windows shells". PowerShell doesn't search the current working directory, and CMD and `CreateProcessW()` don't either if `NoDefaultCurrentDirectoryInExePath` is defined.